### PR TITLE
Feature/nasa cmr

### PR DESCRIPTION
**Pull request by mpu-creare**
Using NASA's CMR instead of looking through the website to construct and index for the SMAP-Sentinel data. 

NOTE: This PR sneaks in a few bugfixes and additional features needed during development and testing
* `ctype` added to clinspace and crange
* Added an xarray Dataset data type -- very useful for reading netcdf files
* Added a return_index option to the unique function of coordinates
* Fixed parts of the `UnitsDataArray` adding a unit test (this will be further fleshed out in #154)

### CODE REVIEW

**Comment by mpu-creare** [comment]
This is to close #210.
@dsully1992 please have a quick look. It would be good to see how the NASA CMR works.

File: podpac/core/coordinates/cfunctions.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from podpac.core.coordinates.uniform_coordinates1d import UniformCoordinates1d
 from podpac.core.coordinates.stacked_coordinates import StackedCoordinates
 
-def crange(start, stop, step, name=None):
+def crange(start, stop, step, name=None, ctype=None):
```
Comment (mpu-creare) [inline podpac/core/coordinates/cfunctions.py]: @jmilloy FYI

File: podpac/core/coordinates/coordinates.py [hunk 2]
```diff
@@ -838,7 +838,7 @@ def intersect(self, other, outer=False, return_indices=False):
         else:
             return Coordinates(intersections)
 
-    def unique(self):
+    def unique(self, return_index=False):
```
Comment (mpu-creare) [inline podpac/core/coordinates/coordinates.py]: @jmilloy, FYI

**Comment by jmilloy** [comment]
@mpu-creare  We've used `return_indices` as the argument in `select` and `intersect`, and you've used `return_index` here. Was that intentional, or should we make those consistent?

**Comment by mpu-creare** [comment]
Thanks for catching that. Please update to make consistent.

**Pull request merged**